src/v/utils/to_string.h

```diff
@@ -12,13 +12,29 @@
 #pragma once
 
 #include "seastarx.h"
+#include "ssx/sformat.h"
 
 #include <seastar/core/lowres_clock.hh>
 #include <seastar/core/print.hh>
 
 #include <optional>
 #include <ostream>
 
+inline ss::sstring human_readable_bytes(uint64_t bytes) {
+    constexpr std::array<std::string_view, 5> units{
+      "B", "KB", "MB", "GB", "TB"};
+    uint64_t rest = bytes;
+    auto unit = 0U;
+    for (; rest >= 1024U && unit < (units.size() - 1); ++unit) {
+        rest = (rest >> 10U);
+    }
+
+    uint64_t single = ((uint64_t)1 << (10U * unit));
+    double d_bytes = 1.0 + (double)(bytes - single) / single;
+
+    return ssx::sformat("{:.2f} {}", d_bytes, units.at(unit));
```

Comment:
if we represent up to full range necessary for uint64_t then we could avoid the runtime out-of-range checks. this might also simplify the algorithm, too because presumably it would be unnecessary to check for `unit < units.size() -1`

Reply:
we have that already here: https://github.com/vectorizedio/redpanda/blob/dev/src/v/utils/human.cc